scrapy/extensions/feedexport.py

```diff
@@ -241,66 +242,115 @@ def __init__(self, crawler):
 
         self.storages = self._load_components('FEED_STORAGES')
         self.exporters = self._load_components('FEED_EXPORTERS')
+        self.storage_batch_item_count = self.settings.getint('FEED_STORAGE_BATCH_ITEM_COUNT')
```

Comment:
Instead of doing this here, please update [`scrapy.utils.conf.feed_complete_default_values_from_settings`](https://github.com/scrapy/scrapy/blob/2.2.0/scrapy/utils/conf.py#L114). That would simplify the `if` at line 339.